resources/models/reservation.py

```diff
@@ -364,12 +364,20 @@ def set_state(self, new_state, user):
                 )
             self.send_reservation_changed_mail_to_user()
             return
+        
+        if new_state == Reservation.REQUESTED:
+            self.requested_at = timezone.now()
+
+        if new_state in (Reservation.CONFIRMED, Reservation.WAITING_FOR_PAYMENT):
+            if old_state == Reservation.REQUESTED:
+                self.approver = user
+                self.approved_at = timezone.now()
```

Comment:
Is there a unit test covering these state changes?

Reply:
Forgot to include them in the PR. I'll add one more test and add them.

Reply:
Tests are included now